wagtail/wagtailcore/models.py

```diff
@@ -108,7 +109,7 @@ def find_for_request(request):
         still be routed to a different hostname which is set as the default
         """
         try:
-            hostname = request.get_host().split(':')[0]
+            hostname, port = split_domain_port(request.get_host())
```

Comment:
Presumably, retrieving `port` here means that we can eliminate the logic below which performs this as a separate step (lines 118-123).